File: uniffi_bindgen/src/bindings/python/templates/UInt32Helper.py
class FfiConverterUInt32(FfiConverterPrimitive):
    @classmethod
    def lower(cls, value):
        if not 0 <= value < 2**32:
            raise ValueError("u32 requires {} <= value < {}".format(0, 2**32))
        return super().lower(value)

    @staticmethod
    def read(buf):
        return buf.readU32()

    @staticmethod
    def write(value, buf):
        buf.writeU32(value)

File: uniffi_bindgen/src/bindings/dart/templates/UInt32Helper.dart
fileprivate struct FfiConverterUInt32: FfiConverterPrimitive {
    typealias FfiType = UInt32
    typealias DartType = UInt32

    public static func read(from buf: inout (data: Data, offset: Data.Index)) throws -> UInt32 {
        return try lift(readInt(&buf))
    }

    public static func write(_ value: DartType, into buf: inout [UInt8]) {
        writeInt(&buf, lower(value))
    }
}
